sdcm/provision/azure/provisioner.py

```diff
@@ -34,38 +36,59 @@
 class AzureProvisioner(Provisioner):  # pylint: disable=too-many-instance-attributes
     """Provides api for VM provisioning in Azure cloud, tuned for Scylla QA. """
 
-    def __init__(self, test_id: str, region: str,  # pylint: disable=unused-argument
-                 azure_service: AzureService = AzureService(), **kwargs):
-        super().__init__(test_id, region)
+    def __init__(self, test_id: str, region: str, availability_zone: str,
+                 azure_service: AzureService = AzureService(), **_):
+        availability_zone = self._convert_az_to_zone(availability_zone)
+        super().__init__(test_id, region, availability_zone)
         self._azure_service: AzureService = azure_service
         self._cache: Dict[str, VmInstance] = {}
         LOGGER.debug("getting resources for %s...", self._resource_group_name)
-        self._rg_provider = ResourceGroupProvider(self._resource_group_name, self._region, self._azure_service)
+        self._rg_provider = ResourceGroupProvider(
+            self._resource_group_name, self._region, self._az, self._azure_service)
         self._network_sec_group_provider = NetworkSecurityGroupProvider(self._resource_group_name, self._region,
                                                                         self._azure_service)
-        self._vnet_provider = VirtualNetworkProvider(self._resource_group_name, self._region, self._azure_service)
+        self._vnet_provider = VirtualNetworkProvider(
+            self._resource_group_name, self._region, self._az, self._azure_service)
         self._subnet_provider = SubnetProvider(self._resource_group_name, self._azure_service)
-        self._ip_provider = IpAddressProvider(self._resource_group_name, self._region, self._azure_service)
+        self._ip_provider = IpAddressProvider(self._resource_group_name, self._region, self._az, self._azure_service)
         self._nic_provider = NetworkInterfaceProvider(self._resource_group_name, self._region, self._azure_service)
-        self._vm_provider = VirtualMachineProvider(self._resource_group_name, self._region, self._azure_service)
+        self._vm_provider = VirtualMachineProvider(
+            self._resource_group_name, self._region, self._az, self._azure_service)
         for v_m in self._vm_provider.list():
             self._cache[v_m.name] = self._vm_to_instance(v_m)
 
+    @staticmethod
+    def _convert_az_to_zone(availability_zone: str) -> str:
+        """When user provides letter instead of zone number, convert it to number."""
```

Comment:
i don't fully understand what it means

Reply:
I added some more info why we need this function. If still it is not clear, let me know.